mongo/errors.go

```diff
@@ -901,3 +903,17 @@ func joinBatchErrors(errs []error) string {
 
 	return buf.String()
 }
+
+// ErrorCodesFrom returns the list of server error codes contained in err.
+func ErrorCodesFrom(err error) []int {
+	if err == nil {
+		return nil
+	}
+
+	var ec interface{ ErrorCodes() []int }
+	if errors.As(wrapErrors(err), &ec) {
+		return ec.ErrorCodes()
```

Comment:
Can users use `errors.As` to convert the error to a `mongo.ServerError` and get the same information?

E.g.
```go
var se mongo.ServerError
if errors.As(err, &se) {
    se.ErrorCodes()
}
```

Reply:
This would only work if `mongo.ServerError` is in the error tree for the arbitrary error which isn't the case for errors propagated by event monitoring. Here's an example: https://github.com/prestonvasquez/go-playground/blob/01e0e77437bdd9ba0a23d912121e152880dc19e9/mgd_errors_test.go#L17

Reply:
@matthewdale The monitoring objects that are passed to clients [are mutable](https://github.com/prestonvasquez/go-playground/blob/eae259bbfd3b48d96a7824283e0dd8f9a00fe110/mgd_client_test.go#L13), so users could easily (albeit probably accidentally) undo any of our own mutation. I agree that we should improve ergonomics for event consumers to use `errors.Is` and `errors.As`, but this should be done at emission rather than mutating user-provided monitors. 

`mongo.ErrorCodes` is clear and resolves the immediate problem at hand. If you'd like, I can a ticket to decouple `wrapErrors` into its own utility. If ever implemented, we can deprecate `mongo.ErrorCodes` in favor of the more idiomatic approach.